src/query/storages/fuse/src/operations/merge_into/processors/processor_merge_into_matched_and_split.rs

```diff
@@ -318,8 +319,39 @@ impl Processor for MatchedSplitProcessor {
                 current_block = op.execute(&self.ctx.get_function_context()?, current_block)?;
                 metrics_inc_merge_into_append_blocks_counter(1);
                 metrics_inc_merge_into_append_blocks_rows_counter(current_block.num_rows() as u32);
+
+                // cornor case: for merge into update, if the target table's column is not null,
+                // for example, target table has three columns like (a,b,c), and we use update set target_table.a = xxx,
+                // it's fine because we have cast the xxx'data_type into a's data_type in `generate_update_list()`,
+                // but for b,c, the hash table will transform the origin data_type (b_type,c_type) into
+                // (nullable(b_type),nullable(c_type)), so we will get datatype not match error, let's transform
+                // them back here.
+                let res_entries = current_block.columns();
```

Comment:
@b41sh need your help to review this cast codes.

Reply:
This cast seems a bit redundant, but there's no better way to do it at the moment, so we can implement it this way first and see if there's a better way to modify it later.